src/lib/ipsec/esp.lua

```diff
@@ -48,6 +48,8 @@ local PAYLOAD_OFFSET = ETHERNET_SIZE + IPV6_SIZE
 local ESP_NH = 50 -- https://tools.ietf.org/html/rfc4303#section-2
 local ESP_SIZE = esp:sizeof()
 local ESP_TAIL_SIZE = esp_tail:sizeof()
+local ESP_RESYNC_THRESHOLD = 4 -- Resync after this many consecutive failed decaps
+local ESP_RESYNC_ATTEMPTS = 6 -- Try this many increments of seq_high
```

Comment:
Some considerations: a) Anyone who can inject or forge packets might be able to trigger an attempt to resynchronize by sending packets with the right (unencrypted) header fields or replaying very old packets.  Every `SP_RESYNC_THRESHOLD`th such packet will trigger resync (as opposed to *every* such packet after the `SP_RESYNC_THRESHOLD`th due to [Line 180](https://github.com/vanfstd/snabb/blob/ef2b4027d3baaabf337e8fb521d3bdd76f994b15/src/lib/ipsec/esp.lua#L180))

Next, the actual cost of a resync operation is roughly `ESP_RESYNC_ATTEMPTS` times the cost of decrypting a packet payload. That's not much in absolute terms, but compared to the overhead of regular packets, it is.  However, the larger the value of `ESP_RESYNC_ATTEMPTS`, the larger a packet loss we can deal with before losing the ability to resynchronize. The number of lost packets we can tolerate is 2^32 times `ESP_RESYNC_ATTEMPTS`.

Based on that, I'd tend to default to a smallish value for `ESP_RESYNC_ATTEMPTS`. Even a value as low as 5 would result in being able to resynchronize after well over a year of network outage, if the sender keeps up a packet rate of 1000 p/s. We could bump it to 6 or even 8 to feel extra sure about it.

Contrary, the value of `ESP_RESYNC_THRESHOLD` should probably default to something substantially bigger. Maybe 1024.  If we are in a situation where we'd resync, we're already past 2^32 lost packets, so it doesn't matter if we require some more to fail before spending resources on resync, making it less attractive to try and abuse resync for DoS purposes in the process.